kong/templates/nginx_kong.lua

```diff
@@ -168,6 +168,12 @@ server {
         proxy_set_header      X-Forwarded-Path   $upstream_x_forwarded_path;
         proxy_set_header      X-Forwarded-Prefix $upstream_x_forwarded_prefix;
         proxy_set_header      X-Real-IP          $remote_addr;
+> for _, name in ipairs(headers_upstream) do
```

Comment:
I checked the code of conf_loader, could we use 
```
> if enabled_headers_upstream["x-kong-request-id"] then
  proxy_set_header X-Kong-Request-Id $kong_request_id;
> end
```

Reply:
We are unable to perform a case-insensitive hash lookup.

Reply:
let me verify if this can work

Reply:
If I configure enabled_headers as 'X-KOnG-REQuest-ID' in the 'kong.conf' file, I have doubts about its functionality.

Reply:
If we track the code of conf_loader, we can find the key of `enabled_headers_upstream` should be `"X-Kong-Request-Id"`, so the code could be `enabled_headers_upstream["X-Kong-Request-Id"]`

Reply:
done